### PR TITLE
Travis cleanup + Markdown Docs

### PR DESCRIPTION
**Pull request by MyNameIsCosmo**
## Travis Cleanup
Travis was cleaned-up, and the scripts were expanded out into their own respective shell files in `~/.travis`. There's no automated deployment *yet*, but there's plenty of examples on it for when the time comes.
- `~/.travis/build.sh`
    - Build the gradle project. This could include tests, debug, and release apks.
- `~/.travis/deploy.sh`
    - A script that handles automatic deployment. No deployment implemented yet. $CURRENT_TAG could be used to tag the release apk
- `~/.travis/emulator.sh`
    - Build and run the emulator. UI tests should be integrated into this.
- `~/.travis/gh-pages.sh`
    - Automatically build and deploy gh-pages on master branch update or tagged release.

[Recent Travis build](https://travis-ci.org/MyNameIsCosmo/qksms/builds/329387517)
    - Build failed on the debug build as the emulator isn't running.

## Markdown Docs
Utilizing [MkDocs](http://www.mkdocs.org/) to bootstrap, serve, and deploy documentation.  
This could house a basic website, developer documentation, FAQ, and API documentation.
[Check out the qksms website on my gh-pages](https://mynameiscosmo.github.io/qksms/)

[Check out the docs/README.md](https://github.com/MyNameIsCosmo/qksms/blob/qksms3/docs/README.md)

## Codacy - Suggestion
Because why not, you should do automated code review and analytics via a service like [Codacy](https://www.codacy.com)
Codacy [seems to support Kotlin](https://blog.codacy.com/shortcuts-to-learning-kotlin-a2d3af535c80) 

[Check out the analytics for mynameiscosmo/qksms](https://www.codacy.com/app/MyNameIsCosmo/qksms/dashboard?bid=6283780)

### CODE REVIEW

**Comment by MyNameIsCosmo** [comment]
I'm currently playing with [Dokka](https://github.com/Kotlin/dokka) to automatically generate javadoc (and the github-flavored markdown of it) for the documentation site.

There's not much (if any) in-code documentation in this project, but I'm sure that will change in the future.
In any case, this is a good feature to have, even if it is currently generating empty docs.

I will be pushing a commit soon for generating the javadoc GFM.

**Comment by MyNameIsCosmo** [comment]
Javadocs are generated using Dokka and formatted to Github Flavored Markdown.

[Check out the JavaDoc on the gh-pages](https://mynameiscosmo.github.io/qksms/qksms/)

Note: It's not *that* pretty. There's a lot of room for improvement, especially with mkdocs v1.0 hopefully coming soon. That said, Kotlin docs are generated automatically.
Haven't tested out the `.travis/gh-pages.sh` yet, but it should automatically deploy the docs on any tag or push to master.

File: docs/mkdocs.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
+site_name: QKSms
```
Comment (moezbhatti) [inline docs/mkdocs.yml]: Can this be updated to QKSMS (all caps)

**Comment by moezbhatti** [comment]
Sweet! Thanks for all the contributions.

Again, really appreciate the support with Travis. These are great changes.

---

Codacy looks like a fantastic service. I've linked it to the project - I'm excited to start using it and try to bring down the number of issues drastically.

---

Some thoughts about the website

For most information about the project like what's found in the Home and Development tabs, I'd like to use the Wiki feature. There isn't much I'm going to need to write, and I think it will suffice to just have all of that kind of information including the README here in github

For a general landing page for normal users, this is something I'd like to design and build from scratch, and it will more than likely not be able to be built using just Markdown - so this is another thing I wouldn't want to use the site for.

I do like the idea of having a website available for quickly navigating documentation however. Do you know if it's possible to have the docs here link directly to the source code?

Perhaps everything relating to the website should be split up into a separate pull request so that we can talk about it in more detail, and then get the other changes merged in the meantime

**Comment by MyNameIsCosmo** [comment]
I agree on your view with the docs.
I was setting them up in a few other repos, and found it to be useful on my projects so far. Wanted to test travis deployment of docs, and figured this would be a good project to do it in.

I'll split the docs up in a separate branch, and utilize it just for API.
The main reason I like the mkdocs build is for the search functionality.

As for linking the docs, there's supposed to be functionality for it under Dokka, but my settings might be messed up.

I'm closing this pull request, and will open 2 separate ones: travis updates, and docs

**Comment by moezbhatti** [comment]
@MyNameIsCosmo awesome, thanks. Much appreciated!

**Pull request closed without merging**